File: experiments/ccomputation/themain.c
#include <stdio.h>
#include <math.h>

double computation_poly(void) {
	double result = 0.0;
	double delta = 1e-9;
	double start = 0.0;
	double stop = 1.0;
	double x = start;
	double y;
	while (x <= stop) {
		y = (x * x + 7.0 * x + 1) / 3.0;
		result += delta * y;
		x += delta;
	}
	return result;
}

double computation_sin(void) {
	double result = 0.0;
	double delta = 1e-8;
	double start = 0.0;
	double stop = M_PI / 2.0;
	double x = start;
	double y;
	while (x <= stop) {
		y = sin(x);
		result += delta * y;
		x += delta;
	}
	return result;
}


int main(int argc, char *argv[]) {
	int n = 0;
	double x = -1;
	if (argc == 2) {
		n = atoi(argv[1]);
	}
	if (n == 0)
		x = computation_poly();
	else if (n == 1)
		x = computation_sin();
		
	printf("%d %f\n", n, x);
}

File: experiments/computation/themain.py
from sys import argv
from math import pi, sin

from parlib.directive import POSIX
#from parlib.rmath import sin #supported only for C, does not bring a speedup

def computation_poly():
	result = 0.0
	delta = 1e-9
	start = 0.0
	stop = 1.0
	x = start
	while x <= stop:
		y = (x * x + 7.0 * x + 1) / 3.0
		result += delta * y
		x += delta
	
	return result

def computation_sin():
	result = 0.0
	delta = 1e-8
	start = 0.0
	stop = pi/2.0
	x = start
	while x <= stop:
		y = sin(x)
		result += delta * y
		x += delta
	
	return result


def main(argv):				
	try:
		n = int(argv[1])
	except:
		n = 0
	if n == 0:
		x = computation_poly()
	elif n == 1:
		x = computation_sin()
	else:
		x = -1
	print n, " ", x

if __name__ == "__main__":
	from sys import argv
	main(argv)

def entry_point(argv):
	if POSIX:
		from parlib.rthreading import init_threads
		init_threads()
	main(argv)
	return 0

def target(*argv):
	return entry_point, None
